airbyte-persistence/job-persistence/src/main/java/io/airbyte/persistence/job/factory/OAuthConfigSupplier.java

```diff
@@ -85,31 +117,21 @@ public JsonNode injectDestinationOAuthParameters(final UUID destinationDefinitio
     }
   }
 
-  private static boolean injectOAuthParameters(final String connectorName,
-                                               final ConnectorSpecification spec,
-                                               final JsonNode oAuthParameters,
-                                               final JsonNode connectorConfig) {
-    if (!hasOAuthConfigSpecification(spec)) {
-      // keep backward compatible behavior if connector does not declare an OAuth config spec
-      MoreOAuthParameters.mergeJsons((ObjectNode) connectorConfig, (ObjectNode) oAuthParameters);
-      return true;
-    }
-    if (!checkOAuthPredicate(spec.getAdvancedAuth().getPredicateKey(), spec.getAdvancedAuth().getPredicateValue(), connectorConfig)) {
-      // OAuth is not applicable in this connectorConfig due to the predicate not being verified
-      return false;
-    }
-    // TODO: if we write a migration to flatten persisted configs in db, we don't need to flatten
-    // here see https://github.com/airbytehq/airbyte/issues/7624
-    boolean result = false;
-    final JsonNode flatOAuthParameters = MoreOAuthParameters.flattenOAuthConfig(oAuthParameters);
+  /**
+   * Gets the OAuth parameter paths as specified in the connector spec and traverses through them
+   */
+  private static void traverseOAuthOutputPaths(final ConnectorSpecification spec,
```

Comment:
Extracted this logic from the existing injection method so it can be re-used for both injection and masking